test/e2e/tests/request-queuing/chainid-check.spec.js

```diff
@@ -0,0 +1,132 @@
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        // Navigate to extension home screen
+        await driver.navigate(PAGES.HOME);
+
+        // Open Dapp One
+        await openDapp(driver, undefined, DAPP_URL);
+
+        await driver.delay(regularDelayMs);
+
+        const chainIdRequest = JSON.stringify({
+          method: 'eth_chainId',
+        });
+
+        const chainIdCheckBeforeConnect = await driver.executeScript(
+          `return window.ethereum.request(${chainIdRequest})`,
+        );
+
+        assert.equal(chainIdCheckBeforeConnect, '0x539'); // 1337
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Network Selector
+        await driver.clickElement('[data-testid="network-display"]');
+
+        // Switch to second network
+        await driver.clickElement({
+          text: 'Ethereum Mainnet',
+          css: 'p',
+        });
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const secondChainIdCheckBeforeConnect = await driver.executeScript(
+          `return window.ethereum.request(${chainIdRequest})`,
+        );
+
+        // before connecting the chainId should change with the wallet
+        assert.equal(secondChainIdCheckBeforeConnect, '0x1');
+
+        // Connect to dapp
+        await driver.findClickableElement({ text: 'Connect', tag: 'button' });
+        await driver.clickElement('#connectButton');
+
+        await driver.delay(regularDelayMs);
+
+        await switchToNotificationWindow(driver);
+
+        await driver.clickElement({
+          text: 'Next',
+          tag: 'button',
+          css: '[data-testid="page-container-footer-next"]',
+        });
+
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+          css: '[data-testid="page-container-footer-next"]',
+        });
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const firstChainIdCallAfterConnect = await driver.executeScript(
+          `return window.ethereum.request(${chainIdRequest})`,
+        );
+
+        assert.equal(firstChainIdCallAfterConnect, '0x1'); // 1337
+
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Network Selector
+        await driver.clickElement('[data-testid="network-display"]');
+
+        // Switch to second network
+        await driver.clickElement({
+          text: 'Localhost 8546',
+          css: 'p',
+        });
+
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+
+        const secondChainIdCall = await driver.executeScript(
+          `return window.ethereum.request(${chainIdRequest})`,
+        );
+        // after connecting the dapp should now have its own selected network
+        // independent from the globally selected and therefore should have changed
```

Comment:
```suggestion
        // independent from the globally selected and therefore should not have changed
```

should not have changed, right?